src/schemas/json/hammerkit.json

```
@@ -0,0 +1,422 @@
+        },
+        "extend": {
+          "$ref": "#/definitions/taskExtend"
+        },
+        "cache": {
+          "$ref": "#/definitions/taskCache"
+        },
+        "continuous": {
+          "$ref": "#/definitions/taskContinuous"
+        },
+        "envs": {
+          "$ref": "#/definitions/taskEnvs"
+        },
+        "needs": {
+          "$ref": "#/definitions/taskNeeds"
+        },
+        "deps": {
+          "$ref": "#/definitions/taskDeps"
+        },
+        "src": {
+          "$ref": "#/definitions/taskSrc"
+        },
+        "generates": {
+          "$ref": "#/definitions/taskGenerates"
+        },
+        "cmds": {
+          "description": "Commands for the current task",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/taskCommand"
+          }
+        },
+        "shell": {
+          "title": "task shell",
+          "description": "Define which shell is used to execute the commands",
+          "type": "string"
+        },
+        "image": {
+          "title": "task image",
+          "description": "Container image for the current task to run commands in",
+          "type": "string"
+        },
+        "ports": {
+          "description": "A list of ports that get exposed on the host from the container <localPort>:<targetPort>",
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "description": "<localPort>:<targetPort>",
+            "type": "string",
+            "pattern": "\\d+:\\d+"
+          }
+        },
+        "mounts": {
+          "description": "File paths that get mounted into the container\nhttps://no0dles.gitbook.io/hammerkit/task/container#adding-mounts",
+          "type": "array",
+          "uniqueItems": true,
+          "items": {
+            "type": "string",
+            "pattern": "\\d+:\\d+"
+          }
+        }
+      }
+    },
+    "taskCommand": {
+      "anyOf": [
+        {
+          "title": "command",
+          "description": "Command to be executed for the current task",
+          "type": "string"
+        },
+        {
+          "title": "task command with path extension",
+          "description": "Command to be executed in sub directory for the current task",
+          "type": "object",
+          "additionalProperties": false,
+          "required": ["cmd"],
+          "properties": {
+            "cmd": {
+              "title": "command",
+              "description": "Command to be executed for the task",
+              "type": "string"
+            },
+            "path": {
+              "title": "command path",
+              "description": "Path below build file path",
+              "$ref": "https://json.schemastore.org/base.json#/definitions/path"
+            }
+          }
+        }
+      ]
+    },
+    "taskDescription": {
+      "title": "task description",
+      "description": "Description for the current task",
+      "type": "string"
+    },
```

```suggestion
    "taskDescription": {
      "description": "Description for the current task",
      "type": "string"
    },
```

Mandatory fix: `title` is not required for non-`object` properties.